### ISSUE TITLE
[Feature Request] Constexpr Tensor

### ISSUE THREAD

**Issue by Giuseppe499** [romeric/Fastor#159]
I would like to have a `constexpr` constructor for the `Tensor` template. With this feature, I could define some constant tensors at compile time.

**Comment by romeric** [comment]
This should be already possible if you compile with `-DFASTOR_ZERO_INITIALISE` flag.

**Comment by Giuseppe499** [comment]
I tried to add the flag `-DFASTOR_ZERO_INITIALISE` using the cmake command `target_link_options(Fastor INTERFACE -DFASTOR_ZERO_INITIALISE)` but i still can't initialize a `constexpr Tensor<>`.
My CMake file is:
```
cmake_minimum_required(VERSION 3.22)
project(RungeKutta)

set(CMAKE_CXX_STANDARD 14)

add_subdirectory(Fastor)
target_link_options(Fastor INTERFACE -DFASTOR_ZERO_INITIALISE)

add_executable(RungeKutta main.cpp Methods/LobattoIIIC.cpp)
target_link_libraries(RungeKutta Fastor)
```

and my main.cpp is:
```
#include <iostream>
#include "Fastor/Fastor/Fastor.h"

using namespace Fastor;

constexpr Tensor<double, 3, 3> A = {
        {1/6, -1/3, 1/6},
        {1/6, 5/12, -1/12},
        {1/6, 2/3, 1/6}
};

int main() {
    Tensor<double,2,2> A, B;
    A.iota();
    B.iota();
    Tensor<double,2,2> C = A%B;
    print("C", C);

    std::cout << "Hello, World!" << std::endl;
    std::cout << 1/6 << std::endl;
    return 0;
}
```

When i compile using CMake i get this error:
```
error: call to non-‘constexpr’ function ‘Fastor::Tensor<T, Rest>::Tensor(const std::initializer_list<std::initializer_list<U> >&) [with U = int; typename std::enable_if<is_primitive_v_<U>, bool>::type <anonymous> = false; T = double; long unsigned int ...Rest = {3, 3}]’
```

**Comment by romeric** [comment]
Just fixed in [6d7216f](https://github.com/romeric/Fastor/commit/6d7216f304d60d92b9b042566d4d6e94c65086a3). Please download the latest copy of Fastor. 

Note that, this is a road less tested (certain expressions cannot really be assigned using `constexpr`) so if you experience further problems open a new issue. I am closing this one.
